mobify/progressive-web-scaffold
# "Cache" data in the redux store, keyed to their URLs

A pattern where slices of state "cache" data keyed to URLs. The currently routed URL is determined by a `selector` value in that slice of state. Skeleton data is keyed to `placeholder` for use during transitions. Also allow PDP to listen to PLP page received to prepopulate some basic data.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-994

## Changes
- Extract some URL-keyed state caching behavior into utility functions
- Refactor PLP to use these utilities
- Implement it in PDP
- Improve PDP propType validation
- Prepopulate PDP with information parsed from PLPs ("shared state")

## How to test-drive this PR
- Visit various PLP and PDPs. 
- Go back to previously visited pages.
- Notice that they load almost instantaneously, due to their data already being in the cache!
